webv2/index/index.go

```diff
@@ -0,0 +1,209 @@
+package index
+
+import (
+	"fmt"
+
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+	"github.com/cloudspannerecosystem/harbourbridge/webv2/session"
+	utilities "github.com/cloudspannerecosystem/harbourbridge/webv2/utilities"
+)
+
+// IndexSuggestion adds redundant index and interleved index  suggestion in schema conversion process for database.
+func IndexSuggestion() {
+
+	sessionState := session.GetSessionState()
+
+	for _, spannerTable := range sessionState.Conv.SpSchema {
+		CheckIndexSuggestion(spannerTable.Indexes, spannerTable)
+	}
+}
+
+//Helpers method for checking Index Suggestion.
+func CheckIndexSuggestion(index []ddl.CreateIndex, spannerTable ddl.CreateTable) {
+
+	redundantIndex(index, spannerTable)
+	interleaveIndex(index, spannerTable)
+}
+
+// redundantIndex check for redundant Index.
+// If present adds IndexRedandant as an issue in Issues.
+func redundantIndex(index []ddl.CreateIndex, spannerTable ddl.CreateTable) {
+
+	for i := 0; i < len(index); i++ {
+
+		keys := index[i].Keys
+		var primaryKeyFirstColumn string
+		pks := spannerTable.Pks
+
+		for i := range pks {
+			if pks[i].Order == 1 {
+				primaryKeyFirstColumn = pks[i].Col
+			}
+		}
+
+		indexFirstColumn := index[i].Keys[0].Col
+
+		if primaryKeyFirstColumn == indexFirstColumn {
+			columnname := keys[i].Col
+			sessionState := session.GetSessionState()
+			schemaissue := sessionState.Conv.Issues[spannerTable.Name][columnname]
+			schemaissue = append(schemaissue, internal.IndexRedandant)
+			sessionState.Conv.Issues[spannerTable.Name][columnname] = schemaissue
+		}
+
+	}
+}
+
+// interleaveIndex suggests if an index can be converted to interleave.
+// If possible it gets added as a suggestion.
+func interleaveIndex(index []ddl.CreateIndex, spannerTable ddl.CreateTable) {
+
+	//Suggestion gets added only if the table can be interleaved.
+	isInterleavable := spannerTable.Parent != ""
+
+	if isInterleavable {
+
+		for i := 0; i < len(index); i++ {
+			var primaryKeyFirstColumn string
+			pks := spannerTable.Pks
+			for i := range pks {
+				if pks[i].Order == 1 {
+					primaryKeyFirstColumn = pks[i].Col
+				}
+			}
+
+			indexFirstColumn := index[i].Keys[0].Col
+
+			sessionState := session.GetSessionState()
+
+			//Ensuring it is not a redundant index.
```

Comment:
For comments please follow the pattern -

- Start with // followed by a space
- End with a fullstop